CMakeLists.txt

```diff
@@ -17,13 +17,23 @@ if(NOT COMMAND add_llvm_library)
         # add -std=gnu++14
         set(CMAKE_CXX_EXTENSIONS ON)
 
-        set(COMMON_FLAGS "-fPIC -Werror")
+        set(COMMON_FLAGS "-fPIC -Werror -Wall -Wextra -Wno-unused-parameter -Wno-unused-variable")
+
+        if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+            # SVF/lib/Util/TypeBasedHeapCloning.cpp:441:14: error: variable 'clone' is used
+            # uninitialized whenever 'if' condition is false
+            set(COMMON_FLAGS "${COMMON_FLAGS} -Wno-sometimes-uninitialized")
+        elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+            set(COMMON_FLAGS "${COMMON_FLAGS} -Wno-maybe-uninitialized -Wno-sign-compare -Wno-unused-but-set-variable -Wno-cast-function-type -Wno-implicit-fallthrough")
+        endif()
+
```

Comment:
Why should we add these options for compiling the code base. The ideal one should be `-Werror`?

Reply:
Actually, there are a lot of issues in the code if we turn on `-Wextra`.

I  think these reports from the compiler help identifying possible issues a lot.
Those `-Wno-xxx` issues are ones we can have a look at and try to solve.

Reply:
Especially those related reported by `Werror=maybe-uninitialized`. These are potential sources of bugs very hard to debug.